### PR TITLE
Ch dictionary rest endpoint

### PR DESCRIPTION
**Pull request by IvanIvanoff**
#### Summary

[//]: # (#### Related PRs)

[//]: # (#### Additional deploy notes)

[//]: # (#### Screenshots)

### CODE REVIEW

File: lib/sanbase_web/controllers/project_data_controller.ex [hunk 1]
```diff
@@ -0,0 +1,49 @@
+defmodule SanbaseWeb.ProjectDataController do
+  use SanbaseWeb, :controller
+
+  alias Sanbase.Model.Project
+  require Logger
+
+  # Used in production mode to serve the reactjs application
+  def data(conn, _params) do
+    {:ok, data} = Sanbase.Cache.get_or_store({__MODULE__, __ENV__.function}, &get_data/0)
```
Comment (valo) [inline lib/sanbase_web/controllers/project_data_controller.ex]: For how long the data will be cached?

Comment (IvanIvanoff) [reply lib/sanbase_web/controllers/project_data_controller.ex]: 5 minutes

File: lib/sanbase_web/controllers/project_data_controller.ex [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          case project do
+            %{infrastructure: %{code: infr_code}} -> infr_code
+            _ -> nil
+          end
+
+        {:ok, github_organizations} = Project.github_organizations(project)
+
+        {contract, decimals} =
+          case Project.contract_info(project) do
+            {:ok, contrat, decimals} -> {contrat, decimals}
```
Comment (tspenov) [inline lib/sanbase_web/controllers/project_data_controller.ex]:
```suggestion
            {:ok, contract, decimals} -> {contract, decimals}
```

File: lib/sanbase_web/controllers/project_data_controller.ex [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    conn
+    |> put_resp_header("content-type", "application/json; charset=utf-8")
+    |> Plug.Conn.send_resp(200, data)
+  end
+
+  defp get_data() do
+    data =
+      Project.List.projects(
+        preload?: true,
```
Comment (tspenov) [inline lib/sanbase_web/controllers/project_data_controller.ex]: It should be true by default :?

Comment (IvanIvanoff) [reply lib/sanbase_web/controllers/project_data_controller.ex]: I've added this to be explicit

Comment (tspenov) [reply lib/sanbase_web/controllers/project_data_controller.ex]: IMO such explicitness is not needed - passing list of preloads implies that preload is true

**Pull request merged**